homeassistant/components/logger/helpers.py

```diff
@@ -216,3 +217,11 @@ async def async_get_levels(self, hass: HomeAssistant) -> dict[str, int]:
                 )
 
         return dict(combined_logs)
+
+
+get_logger = lru_cache(maxsize=256)(logging.getLogger)
```

Comment:
Log level can be changed from UI, where do we reset the cache?

Reply:
There is no need to reset the cache as the level is set on the object itself. 

It still works as expected and is covered by the existing tests

Reply:
Since lru_cache is thread-safe I think the worst outcome you could have here is two threads creating the logger at the exact same time but since they both would get the lock inside the LRU you still get the same one but both would be a cache miss.  Thats seems extremely rare though and in practice the LRU has a fantastic hit rate

Reply:
https://github.com/python/cpython/blob/e5d0316f35a62df05e1b196689f6998662631836/Lib/logging/__init__.py#L1367

Reply:
I think the cpython implementation could be rewritten to avoid the lock when the logger already exists for the common case

```
diff --git a/Lib/logging/__init__.py b/Lib/logging/__init__.py
index bcee2bab0b..01b1ced083 100644
--- a/Lib/logging/__init__.py
+++ b/Lib/logging/__init__.py
@@ -1335,9 +1335,10 @@ def getLogger(self, name):
         logger and fix up the parent/child references which pointed to the
         placeholder to now point to the logger.
         """
-        rv = None
         if not isinstance(name, str):
             raise TypeError('A logger name must be a string')
+        if (rv := self.loggerDict.get(name)) and not isinstance(rv, PlaceHolder):
+            return rv
         _acquireLock()
         try:
             if name in self.loggerDict:
```

Reply:
I can understand why cpython wouldn't do that as it might be the bulk of cases `getLogger` is a cache miss and its only called frequently where it would be a cache hit for our case

Reply:
I did a self pull to verify the that above behavior is safe against the cpython test suite https://github.com/bdraco/cpython/pull/1